python/cusignal/convolution/convolve.py

```diff
@@ -299,19 +304,38 @@ def fftconvolve(in1, in2, mode="full", axes=None):
     fslice = tuple([slice(sz) for sz in shape])
 
     if not complex_result:
-        sp1 = cp.fft.rfftn(in1, fshape, axes=axes)
-        sp2 = cp.fft.rfftn(in2, fshape, axes=axes)
-        ret = cp.fft.irfftn(sp1 * sp2, fshape, axes=axes)[fslice].copy()
+        if (str(fshape), str(axes), "R2C") in _cupy_fft_cache:
+            rplan = _cupy_fft_cache[(str(fshape), str(axes), "R2C")]
+        else:
+            rplan = _cupy_fft_cache[
+                (str(fshape), str(axes), "R2C")
+            ] = fftpack.get_fft_plan(in1, fshape, axes=axes, value_type="R2C")
+
+        if (str(fshape), str(axes), "C2R") in _cupy_fft_cache:
+            irplan = _cupy_fft_cache[(str(fshape), str(axes), "C2R")]
+        else:
+            irplan = _cupy_fft_cache[
+                (str(fshape), str(axes), "C2R")
+            ] = fftpack.get_fft_plan(in1, fshape, axes=axes, value_type="C2R")
+
+        with rplan:
+            sp1 = cp.fft.rfftn(in1, fshape, axes=axes)
+            sp2 = cp.fft.rfftn(in2, fshape, axes=axes)
+        with irplan:
+            ret = cp.fft.irfftn(sp1 * sp2, fshape, axes=axes)[fslice].copy()
```

Comment:
Out of curiosity, how much of the ~5x perf gain is from cached FFTs?

Reply:
A bulk of the time for `cwt`... Maybe 50%

Reply:
We also see improvements for all `fftconvolve` usage. But I'll show that in a different PR.